ssv/validator_registration.go

```diff
@@ -13,6 +13,7 @@ import (
 
 type ValidatorRegistrationRunner struct {
 	BaseRunner *BaseRunner
+	GasLimit   uint64
```

Comment:
A design question..
Why isn't `FeeRecipientAddress` here?
Why is it part of `Share`?

Also imo `NewValidatorRegistrationRunner` should initialize both `GasLimit` and `FeeRecipientAddress`. If `FeeRecipientAddress` isn't fed a value an error should occur. If `GasLimit` limit isn't fed a value we should use the default 30 mil.

Reply:
Also please adjust `testingutils.TestingValidatorRegistration` to use the correct GasLimit and make sure tests pass

Reply:
> Why is it part of `Share`?

Because it's set in `FeeRecipientAddressUpdatedEvent`, but we don't allow configuring gas limit in contracts, so `GasLimit` is not a part of `Share`.

> or maybe never feed `GasLimit` a value.

Do you mean we should probably just use a 30M constant an remove `GasLimit` from `ValidatorRegistrationRunner`?

Reply:
Regarding `FeeRecipientAddress`:
I am guessing (correct me if I am wrong) that this is a per validator setting.
So why it should be in Share?

Also the comment says:
> // Share holds all info about the QBFT/ SSV Committee for msg signing and verification

So it looks not in place

Reply:
Regarding `GasLimit`

see https://github.com/bloxapp/SIPs/pull/25/files#r1194886554

I think constant is good enough!
Also it will remove all the changed roots hopefully making this PR safer to merge :-)

Reply:
> Also it will remove all the changed roots

Unfortunately not 🙁 They were calculated with `GasPrice == 1`. And the implementation runs spec tests on itself to verify itself. So,
- if we leave `GasPrice == 1` in spec, the implementation won't pass spec tests because we cannot use `1` there
- if we change `GasPrice` to `30_000_000` in spec, the roots will be affected, and also if we need to use another value in the future, we'll need to change roots again
- if we make `GasPrice` configurable and use `GasPrice == 1` in spec and `30_000_000` in the implementation (the way it's done in this PR), the roots will be also affected

Reply:
And anyway, when implementing State Comparison, we'll need to touch roots

Reply:
It is correct to have it as a constant regardless imo.. but if you disagree you can comment on the SIP

I think it is best to have the last commit change the value of the constant and the roots of the tests.
This way we will have a proof to what caused the roots to change.

Reply:
> I am guessing (correct me if I am wrong) that this is a per validator setting.

Yes.

> So why it should be in Share?

Because `Share` describes a validator being a part of a committee. And `FeeRecipientAddress` is one of its properties. Is your concern about the same values in some fields for the same validator for different operators?

> So it looks not in place

Do you mean that the `for msg signing and verification` part does not provide a full description of what `Share` does or anything else?

Reply:
> It is correct to have it as a constant regardless imo.. but if you disagree you can comment on the SIP

I don't mind having it as a constant, I'll do that

Reply:
> Because Share describes a validator being a part of a committee.
AFAIU:
Share describes an operator ~~validator~~ being a part of a committee, and what the operator needs for signing and validation.

`FeeRecipientAddress` is a validator property not an operator.
`Grafitti` for example is an operator property because it can be different for each operator.
`ValidatorPubKey` - is a validator property but at least the operator has to know about it for signing and verification.
`Commitee`, `Quorum`, and `DomainType` - ssv data that makes sense it is in share.

But I can't understand why `FeeRecipientAddress` is there.
But before we do anything let's see what @y0sher thinks.

Reply:
> Share describes a validator being a part of a committee.

I meant that `Share` describes a validator being a part of a committee of operators (in other words, validators' share in the committee of operators). Operators have shares for each of their validators. So it's kind of a descriptor of validator that is used by operator. And I think `FeeRecipientAddress` should be there because it's a validator's property used by operator.

> But before we do anything let's see what @y0sher thinks.

Alright, also if @moshe-blox has any comments to this topic, that would be helpful

Reply:
Fee recipient in shares is inherited from validators which inherit from their owner.
```
- Owner <preferred fee recipient>
  - Validator
    - Share
    - Share
    - Share
    - Share
  - Validator
    - Share
    - Share
    - Share
    - Share
```
I agree that since it cannot change on a per-validator or per-share basis, it'd be ideal if it wouldn't be a property in shares to avoid confusion.

However, changing this would be out of scope for this PR, I think.

@alonmuroch or @lior-blox wdyt?

Reply:
Agree it is out of scope of this PR.
Opened #242